server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -201,6 +202,13 @@ public DateTime(StreamInput in) throws IOException {
             String zoneId = in.readString();
             this.timeZone = ZoneId.of(zoneId);
             this.resolution = DateFieldMapper.Resolution.ofOrdinal(in.readVInt());
+            if (in.getVersion().onOrAfter(Version.V_7_7_0) && in.getVersion().before(Version.V_8_0_0)) {
```

Comment:
I think the `in.getVersion().onOrAfter(Version.V_7_7_0)` can be removed, but I can understand keeping it for clarity and safety.

Reply:
you are right, we don't expect mix cluster to work with versions before 7.7. The intention feels cleaner with it to me. I will keep it